src/Symfony/Component/Cache/CacheItem.php

```diff
@@ -100,6 +100,30 @@ public function expiresAfter($time)
     }
 
     /**
+     * Validates a cache key according to PSR-6.
+     *
+     * @param string $key The key to validate.
+     *
+     * @return string $key if it is valid.
+     *
+     * @throws InvalidArgumentException When $key is not valid.
+     */
+    public static function validateKey($key)
```

Comment:
returning a boolean would mean that it should not throw the exception either, which would complicate things a lot (especially to give the right error message). So this looks like a bad idea to me.